### PR TITLE
Implemented 9, 10 functionalities

### PR DESCRIPTION
**Pull request by benhines1119**
-implemented listOutstandingOrders() functionality 
-implemented printTransactions() functionality
-added getAllOrders() to GroceryStore
-added setOrderFields() to DataTransfer
-added setTranasctionFields() to DataTranser
-added setItemFields() to DataTransfer
-added several fields, getters and setters to DataTransfer
-added memberId to Transaction, getter and setter

### CODE REVIEW

File: src/store/collections/TransactionsList.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 package store.collections;
 
-public class TransactionsList {
-
-}
+public class TransactionsList{
+	
+}
```
Comment (carterclark) [inline src/store/collections/TransactionsList.java:5]: Not sure if he will nick us on this but for coding format there should be a space right here

File: src/ui/UserInterface.java [hunk 2]
```diff
@@ -545,10 +546,177 @@ public void getMemberInfo() {
...
+					if(result.getMemberId().equals(userInput)) {
+						memberId = userInput;
+						member = result;
+						System.out.println("Member found.");
+						//We found a member matching user input, so stop looking for others.
+						sentinel = true;
+						break;
+					}
+					if(!counter.hasNext()) {
+						//If you've managed to get here, there's no matching member.
```
Comment (carterclark) [inline src/ui/UserInterface.java:581]: These comments are really useful

File: src/ui/UserInterface.java [hunk 3]
```diff
@@ -545,10 +546,177 @@ public void getMemberInfo() {
...
+		String userInput = "";
+		String memberId = "";
+		Result member = null;
+		Calendar startingDate = Calendar.getInstance();
+		Calendar endingDate = Calendar.getInstance();
+		SimpleDateFormat dateFormat = new SimpleDateFormat("MM/dd/yyyy");
+		
+		//Loop the opportunity to input member ID until valid member ID is found or the user exits.
+		do {
+			System.out.print("Input member ID: ");
```
Comment (carterclark) [inline src/ui/UserInterface.java]: I think we need to explicitly say that the user can enter "END" with each printout from the system.

Also I think there is a get string method already built in? Just to account for the user giving invalid inputs

Comment (benhines1119) [reply src/ui/UserInterface.java]: There is, but Eclipse complained that it "wasn't sure" what I was calling .equals() on was a String or not. The toString() can be removed, but I always assume an IDE has a better handle on code safety than myself.

Comment (carterclark) [reply src/ui/UserInterface.java]: Oh no, I mean a get string method made by pavil in the UserInterface. Line 75 in the user interface class

Comment (benhines1119) [reply src/ui/UserInterface.java]: Oh yeah didn't see that. I'll change how I handle input.

**Review by paveldanek — COMMENTED** [review]
Hey Ben, for functionalities 9 and 10, there was already a method in Member class (getTransactions()) and OrdersList class (inside the GroceryStore - outstanding())...

**Comment by benhines1119** [comment]
> 
> 
> Hey Ben, for functionalities 9 and 10, there was already a method in Member class (getTransactions()) and OrdersList class (inside the GroceryStore - outstanding())...

Yeah, but I'd have to call it on a Member object in the UI, which would expose that Member object to the UI. I'm trying to avoid that by passing a Result holding member data to a method that gets member transactions, then returns what it finds. If I can avoid importing/using any of our back-end objects in the UI, it's probably for the best.

**Comment by paveldanek** [comment]
Yes, you're right. I looked into it a little further and I realized that more would have to be done, in order to make it work. Your coat looks pretty good, I just wish that we could extract some of it and make it a helper method. In order to reuse code.

**Pull request closed without merging**